src/Microsoft.Health.Fhir.Api/Features/SMART/SmartClinicalScopesMiddleware.cs

```diff
@@ -45,38 +47,58 @@ public async Task Invoke(HttpContext context, RequestContextAccessor<IFhirReques
             {
                 var fhirRequestContext = fhirRequestContextAccessor.RequestContext;
                 var principal = fhirRequestContext.Principal;
-                fhirRequestContext.AccessControlContext.ApplyFineGrainedAccessControl = true;
+                var roles = principal.FindAll(authorizationConfiguration.RolesClaim).Select(r => r.Value);
 
-                // examine the scopes claim for any SMART on FHIR clinical scopes
-                DataActions permittedDataActions = 0;
-                foreach (Claim claim in principal.FindAll(authorizationConfiguration.ScopesClaim))
+                // Only read and apply SMART clinical scopes if the SmartUserRole is present
+                if (roles.Contains(authorizationConfiguration.SmartUserRole))
                 {
-                    var matches = ClinicalScopeRegEx.Matches(claim.Value);
-                    foreach (Match match in matches)
-                    {
-                        fhirRequestContext.AccessControlContext.ClinicalScopes.Add(match.Value);
+                    fhirRequestContext.AccessControlContext.ApplyFineGrainedAccessControl = true;
 
-                        var id = match.Groups["id"]?.Value;
-                        var resource = match.Groups["resource"]?.Value;
-                        var accessLevel = match.Groups["accessLevel"]?.Value;
+                    var fhirUser = principal.FindFirstValue(authorizationConfiguration.FhirUserClaim);
+                    try
+                    {
+                        fhirRequestContext.AccessControlContext.FhirUserClaim = new System.Uri(fhirUser);
```

Comment:
Does this support relative to the server `fhirUser` values like `Patient/123`? Both relative and absolute URLs are in spec.

https://build.fhir.org/ig/HL7/smart-app-launch/scopes-and-launch-context.html#scopes-for-requesting-identity-data

Reply:
Updated to allow relative URLs.